KamerConnect.View.MAUI/Views/HomePreferencesForm.xaml.cs

```diff
@@ -1,15 +1,70 @@
+using KamerConnect.Models;
+using KamerConnect.Services;
+using Npgsql;
+
 namespace KamerConnect.View.MAUI.Views;
 
 public partial class HomePreferencesForm : ContentView
 {
-	public HomePreferencesForm()
+	private readonly PersonService _personService;
+	private Person? _currentPerson;
+
+	public HomePreferencesForm(PersonService personService, Person person)
 	{
+		_personService = personService;
+		_currentPerson = person;
+		CheckIfHousePreferencesFieldsAreNull(person);
+		BindingContext = _currentPerson.HousePreferences;
 		InitializeComponent();
 	}
 
-    private void Button_Clicked(object sender, EventArgs e)
-    {
+	private void Button_Update_house_preferences(object sender, EventArgs e)
+	{
+		ValidateIfAccountDetailsChanged(_currentPerson);
+		CheckIfHomeTypeIsPicked(_currentPerson);
+	}
 
+	public void ValidateIfAccountDetailsChanged(Person person)
+	{ 
+		if (_currentPerson.HousePreferences.SurfaceArea != surfaceEntry.DefaultText)
+		{
+			_currentPerson.HousePreferences.SurfaceArea = surfaceEntry.DefaultText;
+		}
+		if (_currentPerson.HousePreferences.Budget != priceEntry.DefaultText)
+		{
+			_currentPerson.HousePreferences.Budget = priceEntry.DefaultText; 
+		}
+	}
+	
+    private void CheckIfHousePreferencesFieldsAreNull(Person person)
```

Comment:
Kan housePreference niet gewoon null blijven, dat je op de plek waar je ermee werkt die null afvangt. Als je dit straks in de database gooit moet krijg je lege strings in de database en geen null.

Reply:
dit werkte heel slecht met die values te comparen van de fields, Daarom had ik ze preventief op lege strings gezet. Maar ik zal nog kijken of dit lukt als ik nog tijd over houdt. Is het probelamtisch als de fields op lege strings staan ipv null?

Reply:
Het comparen van de fields is als het goed is helemaal niet nodig. Je kan op basis van het formulier een object aanmaken en die in de Update gooien. Checken welke velden veranderd zijn en het aanmaken van een project met null waardes hoeft dan allemaal niet als het goed is.